### PR TITLE
feat: upload: add fullwidth prop for small and medium dropzone responsiveness

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:

- Adds `fullWidth` prop to enable responsive `Upload.Dropzone` when `UploadSize` is `Small` or `Medium`
- Updates stories to add `fullWidth` control
- Adds Unit Tests
- Refactors buttons to use `variant` rather than old tag pattern

https://github.com/EightfoldAI/octuple/assets/99700808/bedd2e4a-6fce-48b8-bf48-05ff974e0cda

The above video was missing a style fixup to the failed upload list item width - here's what the latest commit looks like:

errorFixup

## JIRA TASK (Eightfold Employees Only):
ENG-68106

## CHANGE TYPE:

- [ ] Bugfix Pull Request
- [x] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [x] I have added unittests for this change

## TEST PLAN:
Pull the PR branch then run `yarn` and `yarn storybook`. Verify the `Upload` stories behave as expected.

### CODE REVIEW

File: src/components/Upload/Upload.tsx [hunk 1]
```diff
@@ -581,18 +584,20 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (
...
                 {renderText()}
               </Stack>
               {renderButton()}
             </Stack>
           )}
           {size === UploadSize.Small && (
-            <Stack direction={'horizontal'} fullWidth justify={'space-evenly'}>
-              {renderIcon()}
-              {renderText()}
+            <Stack direction={'horizontal'} fullWidth justify={'space-between'}>
```
Comment (jhoward-eightfold) [inline src/components/Upload/Upload.tsx:596]: I think you don't need the bindings to assign simple strings? true for above implementations as well
```suggestion
            <Stack direction="horizontal" fullWidth justify="space-between">
```

File: src/components/Upload/Upload.types.tsx [hunk 2]
```diff
@@ -249,6 +249,11 @@ export interface ShowUploadListInterface {
    * @default 'button'
    */
   downloadIconButtonType?: UploadButtonHtmlType;
+  /**
+   * Assigns Upload list 100% width.
+   * @default false
+   */
+  fullWidth?: boolean;
```
Comment (jhoward-eightfold) [inline src/components/Upload/Upload.types.tsx:256]: noticed in your attached video that the error item in the list was not full width like the others.

Comment (jhoward-eightfold) [reply src/components/Upload/Upload.types.tsx:256]: does the user have to specify this or is it fully controlled based on the parent component implementation?

File: src/components/Upload/Upload.types.tsx
Comment (jhoward-eightfold) [inline src/components/Upload/Upload.types.tsx:1]: this file is impressively long for a "single" component

File: src/components/Upload/Upload.types.tsx (hunk 2 above)
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.types.tsx:256]: No, I missed it in css, update pushed. :)

File: src/components/Upload/Upload.types.tsx
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.types.tsx:1]: Ya, it's one of the more complex ones.

File: src/components/Upload/Upload.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.tsx:596]: Ya, bad habit of mine from angular days.

File: src/components/Upload/Upload.tsx [hunk 3]
```diff
@@ -581,18 +584,20 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (
             </Stack>
           )}
           {size === UploadSize.Medium && (
-            <Stack direction={'vertical'} gap={'m'}>
-              <Stack direction={'horizontal'} gap={'xs'}>
+            <Stack direction={'vertical'} flexGap={'m'}>
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.tsx:587]: gap is deprecated.

File: src/components/Upload/Upload.tsx [hunk 4]
```diff
@@ -550,7 +552,7 @@ const InternalUpload: React.ForwardRefRenderFunction<unknown, UploadProps> = (
       </Stack>
     );
     const renderButton = (): JSX.Element => (
-      <PrimaryButton
+      <Button
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.tsx:555]: It is, I've noticed that codecov has problems with abstraction.

File: src/components/Upload/Upload.tsx (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.tsx]: Done

File: src/components/Upload/Upload.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Upload/Upload.tsx:555]: Will look into it, as long as the project coverage is not effected we should be okay.

**Review by jhoward-eightfold — APPROVED** [review]

**Pull request merged**